conda-store-server/conda_store_server/_internal/orm.py

```diff
@@ -173,6 +173,7 @@ def __init__(self, specification, is_lockfile: bool = False):
     name: Mapped[str] = mapped_column(Unicode(255), nullable=False)
     spec: Mapped[dict] = mapped_column(JSON, nullable=False)
     sha256: Mapped[str] = mapped_column(Unicode(255), unique=True, nullable=False)
+    # TODO: change to datetime.datetime.now(datetime.UTC) when python 3.10 is dropped
```

Comment:
Can we make an issue for this instead of inline TODOs? They have a habit of getting lost in the repo if there aren't issues for them...

To mitigate datetime deprecation warnings, we could do something like this at the top of the file (or in a `utils.py`) so that the correct function is called on whatever version is currently running:

```python
import sys

if sys.version_info[:2] > (3, 10):
    from datetime.datetime import now
else:
    from datetime.datetime import utcnow as now
```

Reply:
ooo, I like the suggestion of making this an issue opposed to TODO's. 

For the deprecation warnings, I don't think this snippet is going to quite work. [`datetime.datetime.now`](https://docs.python.org/3/library/datetime.html#datetime.datetime.now) will use the default TZ provided by the machine it's running on if it's not given the timezone parameter. And `utcnow` does not accept any arguments. So on my machine (I'm on Pacific time) I get 
```
>>> datetime.datetime.now()
datetime.datetime(2024, 12, 16, 12, 10, 50, 437991)

>>> datetime.datetime.utcnow()
. . .  deprecation warning removed for brevity . . .
datetime.datetime(2024, 12, 16, 20, 11, 2, 237286)

>>> datetime.datetime.utcnow(datetime.UTC)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: datetime.utcnow() takes no arguments (1 given)
```

Reply:
Ahh, I guess it would have to be something like

```python
import sys

if sys.version_info[:2] > (3, 10):
    from datetime.datetime import now
else:
    import datetime
    def now():
        return datetime.datetime.utcnow(datetime.UTC)
```

But I'm fine with just merging as is if that's preferable.

Reply:
Cool, I'm going to go for a merge so we have less work in the future when we can drop python 3.10